frontend/src/lib/components/PropertyFilters/components/TaxonomicPropertyFilter.tsx

```diff
@@ -144,6 +144,7 @@ export function TaxonomicPropertyFilter({
                         value: newValue || null,
                         operator: newOperator,
                         type: filter?.type,
+                        label: filter?.label,
```

Comment:
The trouble with the `label` approach is that once someone changes the group's key (e.g. "organization" -> "company"), all old filters will be invalid. That's why we fetch the group/action/cohort/etc name in most places.

Reply:
Yes, I'm aware. I decided the benefit significantly outweighed the cost.